### DISCUSSION TITLE
[DEV] What IDE should I use on linux

### DISCUSSION THREAD

**Discussion by wizzymore** [azerothcore/azerothcore-wotlk#3975] [General]
I don't know if it is the correct word, but how can you get the completion data or an IDE to check for errors like Visual Studio on Linux? I don't care that much about showing errors, but autocompletion would be something really really nice

**Comment by wizzymore** [comment]
I found out Visual Code give you something about declarations and someties the competion of that thing so it is kind of okay but still not Visual Studio good

**Comment by Yehonal** [comment]
Visual studio code has an excellent support for c++ intellisense

Other alternative are Eclipse, Kdevelop etc.

Il Gio 21 Mar 2019, 10:18 WiiZZy <notifications@github.com> ha scritto:

**Comment by wizzymore** [comment]
~Wow, KDevelop is actually nice and i really enjoy it, thanks for the suggestion, i think is better than VS Code ( personal opinion )~
Totally wrong, better thing in KDevelop is that i can build and it will run cmake and make the build with my max cores automatically but the auto completion is only locally plus a lot of include errors even after parsing the project. Still, do you know a way to make VS build my project so i don't have to do it by console everytime?
After installing like 3-4 cmake extensions i got it right in the end and now it feels like VS on windows.

EDIT:
For someone new to linux or VS Code like me the way to do this is by opening the repo with File->Open Folder. if you have Git installed it will even tell you the changes, if you want to build it install first 3 CMake extensions and it will work like charm, be sure you have the last version of cmake installed.

**Comment by FrancescoBorzi** [comment]
@wizzymore excellent question.

I'm a GNU/Linux user too and struggled exactly with the same problem when it comes to AC development.

I solved using CLion, which is a great IDE and it's multiplatform (so available also for GNU/Linux and macOS systems):

https://www.jetbrains.com/clion/

The only problem is that CLion is not free, but we got free open source licenses for AzerothCore. You can find more info here: https://github.com/azerothcore/forum/issues/4

Visual Studio Code is also very good, but for C++ is not as good as CLion.

**Comment by wizzymore** [comment]
I'll give a shot to CLion, i will come with an edit for feedback

**Comment by FrancescoBorzi** [comment]
@wizzymore have a look at:

https://stackoverflow.com/questions/tagged/azerothcore+clion

you can find some useful information there

**Comment by wizzymore** [comment]
Okay, so the set-up was pretty easy as i worken in jetbeans apllications, cmake was easy to setup but a notice, it was way way fast. Normal build time 9 mins CLion build time (only cleans) 5 mins and like 10 seconds.
But, however, how do i solver this:
![image](https://user-images.githubusercontent.com/25827787/54751342-55c2cd00-4be3-11e9-972a-aac5febb518b.png)
And how do i get to see the errors of the file

**Comment by FrancescoBorzi** [comment]
@wizzymore I don't have my CLion setup with me at the moment so I'm not sure how to help you. However, is it only about that declaration or any code navigation is not working for you?

For me they usually work fine. Have you tried invalidate cache and restart CLion? (**File** -> ** Invalidate Cache / Restart **).

I think this specific discussion (about how to solve this issue) should be moved to a separate thread (ideally on stackoverflow).

**Comment by wizzymore** [comment]
No, i found the fix. It seems like with the current CMake clang 7 is broken. It couldn't import the include files. Purged clang-7 and installed clang-6.0 fixed the problem, seems to work with this too:
-- The C compiler identification is GNU 8.2.0
-- The CXX compiler identification is GNU 8.2.0

**Comment by wizzymore** [comment]
Let's move to discord to stop spamming, i will make a new thread here on foruming talking about the IDE and wich is good for what and how to setup them.
Add me -> WiZZy#8181

**Comment by FrancescoBorzi** [comment]
>  CMake clang 7 is broken

this should be reported as a bug

**Comment by BarbzYHOOL** [comment]
I personally use Sublimetext but it's not exactly an IDE

**Comment by Psilovybin** [comment]
I'm using Qt Creator in linux..

**Comment by FrancescoBorzi** [comment]
@Psilovybin is it straightforward to setup an AC development environment on Qt Creator?

It would be nice if you could document the setup procedure, for example like I did with CLion :

https://stackoverflow.com/questions/54206799/how-to-run-and-debug-azerothcore-using-clion

**Comment by wizzymore** [comment]
Taking a look back at this conversation after two years, my god how much I've grown over the years.

Thanks so much guys for the start of my carrier, i learned a lot with you.
